packages/amazonq/test/e2e/amazonq/transformByQ.test.ts

```diff
@@ -0,0 +1,347 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import { qTestingFramework } from './framework/framework'
+import sinon from 'sinon'
+import { Messenger } from './framework/messenger'
+import { JDKVersion, TransformationType, transformByQState } from 'aws-core-vscode/codewhisperer'
+import { GumbyController, startTransformByQ, TabsStorage } from 'aws-core-vscode/amazonqGumby'
+import { using, registerAuthHook, TestFolder } from 'aws-core-vscode/test'
+import { loginToIdC } from './utils/setup'
+import { fs } from 'aws-core-vscode/shared'
+import path from 'path'
+
+describe('Amazon Q Code Transformation', function () {
+    let framework: qTestingFramework
+    let tab: Messenger
+
+    before(async function () {
+        await using(registerAuthHook('amazonq-test-account'), async () => {
+            await loginToIdC()
```

Comment:
👍 if these tests are using the actual service, then e2e is the right place for them.